### PR TITLE
Swift Crypto HMAC integration

### PR DESCRIPTION
**Pull request by 0xTim**
Migrate the JWT HMAC functionality to sit on top of Swift Crypto instead of calling directly to BoringSSL.

### CODE REVIEW

**Comment by 0xTim** [comment]
@Lukasa would be good to get a sanity check on the migration to Crypto when you get a chance

File: Sources/JWTKit/Signing/HMAC/HMACSigner.swift [hunk 1]
```diff
@@ -1,35 +1,14 @@
...
-        var count: UInt32 = 0
-
-        guard hash.withUnsafeMutableBytes({
-            return CJWTKitBoringSSL_HMAC_Final(context, $0.baseAddress?.assumingMemoryBound(to: UInt8.self), &count)
-        }) == 1 else {
-            throw JWTError.signingAlgorithmFailure(HMACError.finalizationFailure)
-        }
-        return .init(hash[0..<Int(count)])
+        let authentication = Crypto.HMAC<SHA_TYPE>.authenticationCode(for: plaintext, using: self.key)
+        return Data(authentication).copyBytes()
```
Comment (Lukasa) [inline Sources/JWTKit/Signing/HMAC/HMACSigner.swift]: I think you can skip the intermediate `Data` object and just instantiate the `Array` directly.

Comment (0xTim) [reply Sources/JWTKit/Signing/HMAC/HMACSigner.swift]: Ah good spot

File: Sources/JWTKit/Signing/HMAC/HMACSigner.swift [hunk 2]
```diff
@@ -1,35 +1,14 @@
-import CJWTKitBoringSSL
+import Crypto
+import Foundation
 
-internal struct HMACSigner: JWTAlgorithm {
-    let key: [UInt8]
-    let algorithm: OpaquePointer
+internal struct HMACSigner<SHA_TYPE>: JWTAlgorithm where SHA_TYPE: HashFunction {
```
Comment (tanner0101) [inline Sources/JWTKit/Signing/HMAC/HMACSigner.swift]: Upper snake case is a little weird here. Could this be `SHAType` or `Hash`?

Comment (0xTim) [reply Sources/JWTKit/Signing/HMAC/HMACSigner.swift]: Fixed

**Pull request merged**